src/field_theory/ratfunc.lean

```diff
@@ -87,6 +93,32 @@ lemma to_fraction_ring_injective :
   function.injective (to_fraction_ring : _ → fraction_ring (polynomial K))
 | ⟨x⟩ ⟨y⟩ rfl := rfl
 
+/-- Non-dependent recursion principle for `ratfunc K`: if `f p q : P` for all `p q`,
+such that `p * q' = p' * q` implies `f p q = f p' q'`, then we can find a value of `P`
```

Comment:
There's another side condition on `q` and `q'`, right. I think it should be mentioned in the docstring.